docs/learning/37-cd-pipeline-design.md

```
@@ -17,1000 +17,530 @@ docker compose build
+on:
+  push:
+    branches: [main]
+    paths-ignore:
+      - 'docs/**'
+      - '**.md'
+      - '.claude/**'
+  workflow_dispatch:
+    inputs:
+      force_rebuild:
+        type: boolean
+        default: false
```

markdownlint-cli2 (0.22.0)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)